src/systemctl/systemctl-list-units.c

```diff
@@ -760,3 +761,205 @@ int verb_list_timers(int argc, char *argv[], void *userdata) {
 
         return r;
 }
+
+struct automount_info {
+        const char* machine;
+        const char* id;
+        char* what;
+        char* where;
+        usec_t timeout_idle_usec;
+        bool mounted;
+};
+
+static int automount_info_compare(const struct automount_info *a, const struct automount_info *b) {
+        int r;
+
+        assert(a);
+        assert(b);
+
+        r = strcasecmp_ptr(a->machine, b->machine);
+        if (r != 0)
+                return r;
+
+        return strcmp(a->where, b->where);
+}
+
+static int collect_automount_info(sd_bus* bus, const UnitInfo* info, struct automount_info *ret_info) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message* reply = NULL;
+        _cleanup_strv_free_ char** triggered = NULL;
+        _cleanup_free_ UnitInfo* triggered_infos = NULL;
+        _cleanup_free_ char* where = NULL, *what = NULL, *state = NULL;
+        usec_t timeout_idle_usec;
+        BusLocator locator;
+        int r, n;
+
+        assert(bus);
+        assert(info);
+        assert(ret_info);
+
+        locator = (BusLocator) {
+                .destination = "org.freedesktop.systemd1",
+                .path = info->unit_path,
+                .interface = "org.freedesktop.systemd1.Automount",
+        };
+
+        r = bus_get_property_string(bus, &locator, "Where", &error, &where);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get automount target: %s", bus_error_message(&error, r));
+
+        r = bus_get_property_trivial(bus, &locator, "TimeoutIdleUSec", &error, 't', &timeout_idle_usec);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get idle timeout: %s", bus_error_message(&error, r));
+
+        r = get_triggered_units(bus, info->unit_path, &triggered);
+        if (r < 0)
+                return r;
+
+        n = get_unit_list(bus, NULL, triggered, &triggered_infos, 0, &reply);
```

Comment:
you don't need to list the triggered units here, because automounts only trigger a single automount unit. Hence, add a safety check that `strv_length(triggered)` equals 1, and then you can use the one unit name in the list directly. And then you can use `unit_dbus_path_from_name()` to generate the dbus path for the unit directly client side, without any roundtrip. Thus you can get rid of the for loop below.

Reply:
Right. I've already rewritten this code to address @keszybz's comment.